.github/workflows/unittests.yml

```diff
@@ -0,0 +1,26 @@
+name: Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9', 'pypy-3.6', 'pypy-3.7']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[contrib]
+    - name: Test with pytest
```

Comment:
Wouldn't it be more convenient to also run the tests as Github Actions rather than running a single action with pytest?

Reply:
I'm not quite following. This does run the tests as a GitHub action. Can you clarify what you are asking?

Reply:
I wanted to know if it wouldn't be nicer to have a single action for each unit test bypassing pytest.

Reply:
What do you expect to gain from this (other than additional overhead and extra work when adding new tests)?

Reply:
I'm not sure running each test individually is feasible or practical. We currently have 67 tests (which is rather on the low side).
* With the 6 different Python versions, that would be over 400 actions already.
* We would have to do our own test discovery and create actions from that, at that scale automatically. I'm not quite sure how to do that (GH action writing actions?) nor how to keep it maintainable.

Reply:
Okay maybe in this case the overhead in actions is impracticable. I just wondered, if executing them as actions, would improve the overview, since then there would be a single test instance (actions) taking care. With pytest, you kind of have an additional nested test layer within an action.